tests/test_ec2.py

```diff
@@ -493,6 +494,42 @@ def test_ec2_stop(self):
         self.assertEqual(len(resources), 1)
 
 
+class TestReboot(BaseTest):
+
+    def test_ec2_reboot(self):
+        session_factory = self.replay_flight_data(
+            'test_ec2_reboot')
+        policy = self.load_policy({
+            'name': 'ec2-test-reboot',
+            'resource': 'ec2',
+            'filters': [
+                {'tag:Testing': 'not-null'}],
+            'actions': [
+                {'type': 'reboot'}]},
+            session_factory=session_factory)
+        resources = policy.run()
+        self.assertEqual(len(resources), 2)
+        running = []
+        for i in resources:
+            if i['State']['Name'] == 'running':
+                running.append(i['InstanceId'])
+        if self.recording:
+            time.sleep(25)  
+        instances = utils.query_instances(
+            session_factory(),
+            InstanceIds=[r['InstanceId'] for r in resources])
+
+        cur_running = []
+        for i in instances:
+            if i['State']['Name'] == 'running':
+                cur_running.append(i['InstanceId'])
+
+        cur_running.sort()
+        running.sort()
+
+        self.assertEqual(cur_running, running)
```

Comment:
as a test this is a little weak, ie the reboot action could be a noop and this assertion would still succeed. feels like we need a third assertion here wrt to state before the sleep after the action, to have some notion that state changed.

Reply:
Yeah I was looking for a good indication that shows the reboot occurring or that it worked but I didn't find anything to go off of.  Do you have any ideas?

Reply:
@kapilt What do you think of the idea of using the ec2 get console output command for the test and reading the most recent line and waiting until it returns back to the ready prompt before it passes its reboot test?

Reply:
if you do a describe instances with an ec2 client right before the wait you should see evidence of a state change, ie the instance should be starting its state transition.

Reply:
@kapilt I have tested the describe instance status with an instance while rebooting and the status doesn't change at all which is why I started looking at the Console Output which seems to work but is a little whacky as a test

Reply:
thats fine lets move ahead with it as is.